src/msg/async/rdma/Infiniband.cc

```diff
@@ -890,3 +768,33 @@ const char* Infiniband::qp_state_string(int status) {
     default: return " out of range.";
   }
 }
+
+void Infiniband::handle_pre_fork()
+{
+  device->uninit();
```

Comment:
Correct. This commit is a preparation only. It only initializes the selected device (from the config parameter).
I will soon send you a patch that enable using more than 1 device, and the uninit() will be modified accordingly.